docs/tutorial/durham_reframe_tutorial.md

```diff
@@ -0,0 +1,409 @@
+    self.build_system.cppflags = [f'-DSTREAM_ARRAY_SIZE=$((2 ** {self.arraysize}))']
+```
+
+You can have multiple parameters. ReFrame will run all parameter combinations by default.
+
+---
+
+## [Build systems](https://reframe-hpc.readthedocs.io/en/v4.5.2/tutorial_advanced.html#more-on-building-tests)
+- [Build systems Reference](https://reframe-hpc.readthedocs.io/en/v4.5.2/regression_test_api.html#build-systems)
+
+----
+
+### [Make](https://reframe-hpc.readthedocs.io/en/v4.5.2/tutorial_advanced.html#more-on-building-tests) 
+
+- Tutorial in `tutorials/advanced/makefiles/maketest.py`. 
+
+> you must set the executable attribute of the test, because ReFrame cannot know what is the actual executable to be run. We then set the build system to Make and set the preprocessor flags as we would do with the SingleSource build system.
+
+
+----
+
+### [Autotools](https://reframe-hpc.readthedocs.io/en/v4.5.2/tutorial_advanced.html#adding-a-configuration-step-before-compiling-the-code)
+> It is often the case that a configuration step is needed before compiling a code with make. To address this kind of projects, ReFrame aims to offer specific abstractions for “configure-make” style of build systems. It supports CMake-based projects through the CMake build system, as well as Autotools-based projects through the Autotools build system.
+
+
+- [Automake Hello example](https://github.com/ntegan/amhello)
+
+```python
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+@rfm.simple_test
+class AutoHelloTest(rfm.RegressionTest):
+    valid_systems = ['*']
+    valid_prog_environs = ['*']
+    sourcesdir = 'https://github.com/ntegan/amhello.git'
+    build_system = 'Autotools'
+    executable = './src/hello'
+    prebuild_cmds = ['autoreconf --install .']
+    time_limit = '5m'
+    
+    @sanity_function
+    def assert_hello(self):
+        return sn.assert_found(r'Hello world\!', self.stdout)
+```
+
+----
+
+### [CMake](https://reframe-hpc.readthedocs.io/en/v4.5.2/regression_test_api.html#reframe.core.buildsystems.CMake)
+- [CMake Hello example](https://github.com/jameskbride/cmake-hello-world)
+
+```python
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+
+@rfm.simple_test
+class CMakeHelloTest(rfm.RegressionTest):
+    valid_systems = ['*']
+    valid_prog_environs = ['*']
+    sourcesdir = 'https://github.com/jameskbride/cmake-hello-world.git'
+    build_system = 'CMake'
+    executable = './CMakeHelloWorld'
+    time_limit = '5m'
+    
+    @sanity_function
+    def assert_hello(self):
+        return sn.assert_found(r'Hello, world\!', self.stdout)
+
+```
+
+----
+
+### [Spack](https://reframe-hpc.readthedocs.io/en/v4.5.2/regression_test_api.html#reframe.core.buildsystems.Spack)
+
+- ReFrame will use a user-provided Spack environment in order to build and test a set of specs.
+- Tutorial in `tutorials/build_systems/spack/spack_test.py`
+- In `rfm_job.out` you can see that it
+  - Creates a blank environment
```

Comment:
Those two bullets appear at the same level as the one above. I think they need another space.